CONTRIBUTING.md

```diff
@@ -76,37 +76,66 @@ Check out our dedicated [Bevy Organization document](/docs/the_bevy_organization
 
 ### Classifying PRs
 
-Our merge strategy relies on the classification of PRs into three categories: **trivial**, **uncontroversial** and **controversial**.
-When making PRs, try to split out more controversial changes from less controversial ones, in order to make your work easier to review and merge.
-PRs that are deemed controversial will receive the [`S-Controversial`](https://github.com/bevyengine/bevy/pulls?q=is%3Aopen+is%3Apr+label%3AS-Controversial) label, and will have to go through the more thorough review process.
+Our merge strategy relies on the classification of PRs on two axes:
 
-PRs are trivial if there is no reasonable argument against them. This might include:
+* How controversial are the design decisions
+* How complex is the implementation
 
-* Fixing dead links.
-* Removing dead code or dependencies.
-* Typo and grammar fixes.
+PRs with non-trivial design decisions are given the [`S-Controversial`] label. This indicates that
+the PR needs more thorough design review or an [RFC](https://github.com/bevyengine/rfcs), if complex enough.
 
-PRs are controversial if there is serious design discussion required, or a large impact to contributors or users. Factors that increase controversy include:
+PRs that are non-trivial to review are given the [`D-Complex`] label. This indicates that the PR
+should be reviewed more thoroughly and by people with experience in the area that the PR touches.
+
+When making PRs, try to split out more controversial changes from less controversial ones, in order to make your work easier to review and merge.
+It is also a good idea to try and split out simple changes from more complex changes if it is not helpful for then to be reviewed together.
 
-1. Changes to project-wide workflow or style.
+Some things that are reason to apply the [`S-Controversial`] label to a PR:
+
+1. Changes to a project-wide workflow or style.
 2. New architecture for a large feature.
-3. PRs where a serious tradeoff must be made.
+3. Serious tradeoffs were made.
 4. Heavy user impact.
```

Comment:
Is adding a lot of docs like in #6649 considered "Heavy user impact"? Initially I thought it would be just about changing the public interface too much, so maybe it's not clear.

Reply:
For reference, that PR was mislabeled (on the basis of its current contents), which has since been fixed.

Very high impact docs changes could be controversial, but I would expect that to be an extreme exception.